src/main/java/de/caritas/cob/uploadservice/api/helper/EmailNotificationHelper.java

```diff
@@ -1,52 +1,82 @@
+import de.caritas.cob.uploadservice.api.service.TenantHeaderSupplier;
 import de.caritas.cob.uploadservice.api.service.helper.ServiceHelper;
+import de.caritas.cob.uploadservice.api.tenant.TenantContext;
+import de.caritas.cob.uploadservice.userservice.generated.web.model.NewMessageNotificationDTO;
+import java.util.Optional;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
 import org.springframework.scheduling.annotation.Async;
 import org.springframework.stereotype.Component;
 import org.springframework.web.client.RestClientException;
-import org.springframework.web.client.RestTemplate;
 
 /*
  * Helper for sending an email notification via the MailService
  */
 @Component
 public class EmailNotificationHelper {
 
-  private final RestTemplate restTemplate;
   private final ServiceHelper serviceHelper;
+  private final TenantHeaderSupplier tenantHeaderSupplier;
+  private final de.caritas.cob.uploadservice.userservice.generated.web.UserControllerApi userControllerApi;
 
   @Autowired
-  public EmailNotificationHelper(RestTemplate restTemplate, ServiceHelper serviceHelper) {
-    this.restTemplate = restTemplate;
+  public EmailNotificationHelper(ServiceHelper serviceHelper, TenantHeaderSupplier tenantHeaderSupplier, de.caritas.cob.uploadservice.userservice.generated.web.UserControllerApi userControllerApi) {
     this.serviceHelper = serviceHelper;
+    this.tenantHeaderSupplier = tenantHeaderSupplier;
+    this.userControllerApi = userControllerApi;
   }
 
   /**
    * Send an email via the UserService
-   *
-   * @param rcGroupId
-   * @param userServiceApiSendNewMessageNotificationUrl
+   *  @param rcGroupId
+   * @param currentTenant
    */
   @Async
   public void sendEmailNotificationViaUserService(
-      String rcGroupId, String userServiceApiSendNewMessageNotificationUrl, String accessToken) {
+      String rcGroupId, String accessToken,
+      Optional<Long> currentTenant) {
 
     try {
-      HttpHeaders header = serviceHelper.getKeycloakAndCsrfHttpHeaders(accessToken);
-      NewMessageNotificationDto notificationDto = new NewMessageNotificationDto(rcGroupId);
-      HttpEntity<NewMessageNotificationDto> request = new HttpEntity<>(notificationDto, header);
+      NewMessageNotificationDTO notificationDto = new NewMessageNotificationDTO().rcGroupId(rcGroupId);
+      addDefaultHeaders(userControllerApi.getApiClient(), accessToken, currentTenant);
+      userControllerApi.sendNewMessageNotification(notificationDto);
+      TenantContext.clear();
 
-      restTemplate.exchange(
-          userServiceApiSendNewMessageNotificationUrl, HttpMethod.POST, request, Void.class);
+    } catch (RestClientException ex) {
+      LogService.logUserServiceHelperError(ex);
+    }
+  }
+
+  @Async
+  public void sendEmailFeedbackNotificationViaUserService(
```

Comment:
The only difference I see between "sendEmailNotificationViaUserService" and "sendEmailFeedbackNotificationViaUserService" is the method call of userControllerApi. Maybe we can use one private method for the logic passing the methodreference as a consumer of "userControllerApi" to prevent code duplication?

Reply:
done